src/develop/masks/detail.c

```diff
@@ -495,9 +495,19 @@ gboolean dt_masks_calc_detail_mask(dt_dev_detail_mask_t *details,
                                const float threshold,
                                const gboolean detail)
 {
+  if((details->roi.width <= 0)
+     || (details->roi.height <= 0)
+     || !details->data
+     || (details->hash == 0))
+    return TRUE;
+
   const size_t msize = (size_t) details->roi.width * details->roi.height;
   float *tmp = dt_alloc_align_float(msize);
-  if(!tmp) return TRUE;
+  if(!tmp)
+  {
+    memset(out, 0, msize * sizeof(float));
```

Comment:
`memset(..)` could be replaced by:
`dt_iop_image_fill(out, 0.0f, details->roi.width, details->roi.height, 1);`